### PR TITLE
tests: check for node.is_connected

### PR DESCRIPTION
**Pull request by lwojtow**

### CODE REVIEW

**Comment by ljedrz** [comment]
Good idea; I'd like to request some adjustments:
- use `slice::windows` instead of `::chunks` to be more strict (also, `into_iter` is redundant)
- rename the closure param from `c` to `pair`
- add two extra assertions performing the same check in reverse (`nodes.windows(2).rev().all(...)`)

File: tests/node.rs [hunk 1]
```diff
@@ -45,6 +45,13 @@ async fn node_connect_and_disconnect() {
     wait_until!(1, nodes.iter().all(|n| n.node().num_connected() == 1));
 
     assert!(nodes.iter().all(|n| n.node().num_connecting() == 0));
+    assert!(nodes.windows(2).all(|pair| pair[0]
+        .node()
+        .is_connected(pair[1].node().listening_addr().unwrap())));
+
```
Comment (ljedrz) [inline tests/node.rs:51]:
```suggestion
```

**Comment by ljedrz** [comment]
LGTM with the extra newline removed and squashed :+1:

**Comment by ljedrz** [comment]
Actually, now that I've had more time to take a look at it, these are not needed due to the following:
- `connect_nodes` is already tested in `topologies.rs`
- `Node::num_connected() == 0` already guarantees that there are no connections

Also, doing checks while going in reverse wouldn't work due to the connected address being different than the listening address (my bad).

**Comment by lwojtow** [comment]
That's not the point. There is a check for:
`pub fn is_connected(&self, addr: SocketAddr) -> bool {
        false
}` but not for returning `true`. If you replace current code with returning `true` all tests will pass.

**Comment by ljedrz** [comment]
That's fair (even if it's implied, I guess); I added this check in https://github.com/ljedrz/pea2pea/commit/b0f9cd28f1d1023bea294dd5873d51473cf24741.

**Pull request closed without merging**